data/tutorials/language/1ms_02_dune.md

```diff
@@ -0,0 +1,269 @@
+---
+id: libraries-dune
+title: Libraries With Dune
```

Comment:
Should we also consider packages?

Yesterday I did another stream full of skill issue on my part where I tried to create all three kinds of libraries with dune (the singleton, wrapped, and unwrapped sorts) as a package.

Where I got confused and fumbled was when I wanted to install the package containing the three libraries in another switch and I had trouble seeing how to name the packages so that they show up.

In a sense this topic has some overlap with the OCaml module system, but it also has significant overlap with dune and opam (so kind of feels like it belongs in Platform Tools).

Reply:
There is indeed an overlap. Yes I believe a fourth document is needed on packaging.
1. Units of compilation: module and functors
2. Units of linking & loading: libraries
3. Units of distribution & installation: packages

I wonder if we should rename the "MODULE SYSTEM" section of the "Language" tutorial to reflect on the units listed above